### PR TITLE
change apollo to envelop

### PR DESCRIPTION
**Pull request by PabloSzx**

### CODE REVIEW

File: extensions/graphql/src/graphql.server.ts [hunk 1]
```diff
@@ -146,25 +145,56 @@ export class GraphQLServer extends Context implements Server {
...
+          res: args.response,
+          // This argument doesn't exist
+          connection: undefined,
+        }) as PromiseOrValue<Record<string, any>>;
+      },
+    });
+
+    // What do we do about these options?
+    this.options.middlewareOptions?.disableHealthCheck;
+    this.options.middlewareOptions?.onHealthCheck;
```
Comment (PabloSzx) [inline extensions/graphql/src/graphql.server.ts]: What do we do about these options?

File: extensions/graphql/src/graphql.server.ts [hunk 2]
```diff
@@ -146,25 +145,56 @@ export class GraphQLServer extends Context implements Server {
...
+
+    // I guess this option can be added to envelop/app
+    this.options.middlewareOptions?.cors;
```
Comment (PabloSzx) [inline extensions/graphql/src/graphql.server.ts]: I can add it in envelop/app if it makes sense to do it

File: extensions/graphql/src/graphql.server.ts [hunk 3]
```diff
@@ -146,25 +145,56 @@ export class GraphQLServer extends Context implements Server {
...
 
-    // Create ApolloServerExpress GraphQL server
-    const serverConfig: ApolloServerExpressConfig = {
-      // enable GraphQL Playground
-      playground: true,
-      context: graphqlContextResolver,
-      subscriptions: false,
-      ...this.options.apollo,
+    // Tons of apollo-specific options that I don't think we would be able to do 1-1
+    this.options.apollo;
```
Comment (PabloSzx) [inline extensions/graphql/src/graphql.server.ts]: this is a big one

**Comment by achrinza** [comment]
I wonder if we should be creating a new, separate package (e.g. `@loopback/graphql-envelop`).

1. Although still "experimental", I believe there's already users of `@loopback/graphql` and we should continue to let it mature separately.

2. Since Envelop is also quite architecturally different from Apollo, it may make more sense to do a rewrite as a separate package.

3. This will also mean that we don't have to map every Apollo-specific feature over to Envelop, and work on it more gradually, or provide more ergonomic APIs that are a better-fit for Enevelop.

**Pull request closed without merging**